demisto_sdk/commands/lint/linter.py

```diff
@@ -180,29 +192,17 @@ def _gather_facts(self, modules: dict) -> bool:
             bool: Indicating if to continue further or not, if False exit Thread, Else continue.
         """
         # Looking for pkg yaml
-        yml_file: Optional[Path] = self._pack_abs_dir.glob([r'*.yaml', r'*.yml', r'!*unified*.yml'], flags=NEGATE)
-        if not yml_file:
-            logger.info(f"{self._pack_abs_dir} - Skipping no yaml file found {yml_file}")
-            self._pkg_lint_status["errors"].append('Unable to find yml file in package')
-            return True
-        else:
-            try:
-                yml_file = next(yml_file)
-            except StopIteration:
-                return True
-        # Get pack name
-        self._pack_name = yml_file.stem
         log_prompt = f"{self._pack_name} - Facts"
-        self._pkg_lint_status["pkg"] = yml_file.stem
-        logger.info(f"{log_prompt} - Using yaml file {yml_file}")
+        self._pkg_lint_status["pkg"] = self._pack_name
+        logger.info(f"{log_prompt} - Using yaml file {self._yml_file}")
         # Parsing pack yaml - in order to verify if check needed
         try:
 
             script_obj: Dict = {}
-            yml_obj: Dict = YAML_Handler().load(yml_file)
+            yml_obj: Dict = YAML_Handler().load(self._yml_file)
```

Comment:
For consistency with other files, please instantiate `yaml = YAML_Handler()` at the top of the file instead of here

Reply:
There is a reason for this.
If we use the same instance for a couple of hours, the instance is out of memory, and we get an exception.
I will add a comment.